metrics/ci_integrations/test/destination/firestore/adapter/firestore_destination_client_adapter_test.dart

```diff
@@ -273,14 +324,267 @@ void main() {
     );
 
     test(
-      ".fetchBuildsWithStatus() returns an empty list",
+      ".fetchBuildsWithStatus() throws an ArgumentError if the given project id is null",
+      () {
+        expect(
+          () => adapter.fetchBuildsWithStatus(null, BuildStatus.successful),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() throws an ArgumentError if the given build status is null",
+      () {
+        expect(
+          () => adapter.fetchBuildsWithStatus(testProjectId, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() throws a DestinationError if fetching a project with the given project id fails",
+      () {
+        whenFetchProject().thenAnswer((_) => Future.error(firestoreException));
+
+        final result = adapter.fetchBuildsWithStatus(
+          testProjectId,
+          BuildStatus.successful,
+        );
+
+        expect(result, throwsDestinationError);
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() throws an ArgumentError if the project with the given project id does not exist",
+      () {
+        whenCheckProjectExists().thenReturn(false);
+
+        final result = adapter.fetchBuildsWithStatus(
+          testProjectId,
+          BuildStatus.successful,
+        );
+
+        expect(result, throwsArgumentError);
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() references the 'build' collection",
+      () async {
+        whenFetchBuildsWithStatus().thenAnswer((_) => Future.value([]));
+
+        await adapter.fetchBuildsWithStatus(
+          testProjectId,
+          BuildStatus.successful,
+        );
+
+        verify(_firestoreMock.collection('build')).called(once);
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() filters builds with the given project id",
       () async {
+        whenFetchBuildsWithStatus().thenAnswer((_) => Future.value([]));
+
+        await adapter.fetchBuildsWithStatus(
+          testProjectId,
+          BuildStatus.successful,
+        );
+
+        verify(
+          _collectionReferenceMock.where('projectId', isEqualTo: testProjectId),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() filters builds with the given build status",
+      () async {
+        const buildStatus = BuildStatus.inProgress;
+        whenFetchBuildsWithStatus(
+          withBuildStatus: buildStatus,
+        ).thenAnswer((_) => Future.value([]));
+
+        await adapter.fetchBuildsWithStatus(
+          testProjectId,
+          buildStatus,
+        );
+
+        verify(
+          _collectionReferenceMock.where(
+            'buildStatus',
+            isEqualTo: '$buildStatus',
```

Comment:
```suggestion
            isEqualTo: '$buildStatus'
```
I'm wondering does it fits?